src/nvim/os/env.c

```diff
@@ -470,9 +481,35 @@ void init_homedir(void)
         EMSG(_(e_prev_dir));
       }
     }
+  }
 #endif
-    homedir = xstrdup(var);
+
+  // Fall back to current working directory if home is not found
```

Comment:
This runs for both Unix and Windows. Function description seems to suggest that this happens in Unix only.

Reply:
Made a slight modification to the current algorithm. For windows, fall back to C, and for Unix, fall back to the current working directory.